tests/test_implementations/test_semsimian_implementation.py

```diff
@@ -161,6 +161,28 @@ def test_similarity_with_custom_ic_map(self):
                     else:
                         raise ValueError(f"Did not get similarity for got {s} and {o}")
 
+    def test_all_by_all_similarity_with_custom_ic_map(self):
+        adapter = self.oi
+
+        adapter.custom_ic_map_path = TEST_IC_MAP.as_posix()
+
+        if not isinstance(adapter, SemanticSimilarityInterface):
+            raise AssertionError("SemanticSimilarityInterface not implemented")
+        entities = [VACUOLE, ENDOMEMBRANE_SYSTEM]
+
+        sim = (adapter.all_by_all_pairwise_similarity(entities, entities, predicates=self.predicates))
```

Comment:
not related to this PR but the way this test uses object variables for constants is a little unusual